Cherry-picked commits for WordPress 6.4 RC3

SiobhyB opened the pull request:
Latest commits for WordPress 6.4 RC3.

Included changes:

* https://github.com/WordPress/gutenberg/pull/55553
* https://github.com/WordPress/gutenberg/pull/55539
* https://github.com/WordPress/gutenberg/pull/55667
* https://github.com/WordPress/gutenberg/pull/55669

Merged/cherry-picked directly to `wp/6.4` for inclusion in RC3, separate to this PR:

* https://github.com/WordPress/gutenberg/pull/55604
* https://github.com/WordPress/gutenberg/pull/55676

glendaviesnz committed:
Patterns: set context of pattern category api call to view so authors and contributors can access (#55553)

DAreRodz committed:
Query Loop: Disallow "enhanced pagination" with core blocks that may contain third-party blocks (#55539)

* Update useContainsThirdPartyBlocks and texts

* Improve texts a bit

* Show the list of unsupported blocks

* Use the BlockTitle component instead

* Simplify modal text

DAreRodz committed:
Query Loop: Disallow "enhanced pagination" with core blocks that may contain third-party blocks (#55539)

* Update useContainsThirdPartyBlocks and texts

* Improve texts a bit

* Show the list of unsupported blocks

* Use the BlockTitle component instead

* Simplify modal text

karmatosed reviewed (approved):
My only note is that on #55539 it might be good to be clearer why it's a candidate, the other issue has much more clarity which is good for committer that will review this.

SiobhyB committed:
Revert "Query Loop: Disallow "enhanced pagination" with core blocks that may contain third-party blocks (#55539)"

This reverts commit f2dd6de59ecd1cb142b5d0b1e7db77d14eb87faf.

SiobhyB commented:
~To tie up all the ends: https://github.com/WordPress/gutenberg/pull/55539 has been reverted as per the summary at https://github.com/WordPress/gutenberg/pull/55539#issuecomment-1783172761.~ 

The reasoning for including #55539 was discussed via Slack and summarised at https://github.com/WordPress/gutenberg/pull/55539#issuecomment-1783236901.

DAreRodz committed:
Revert CSS changes to show/hide embedded pdf (#55667)

SiobhyB committed:
Ensure Term Description block is available in core (#55669)

Reverts changes in https://github.com/WordPress/gutenberg/pull/51053, to address a regression in which the Term Description block was only available in core when the Gutenberg plugin was active.
